### PR TITLE
Allow access to named Parts shape, Field shape and Items from zone 

### LINKED ISSUE #3167: Add accessor for items in shapes
Right now within a Content template `Model.Content.Named("ThePartName")` works but not `Model.Content.ThePartName` as it is documented.

### PR DESCRIPTION
**Pull request by ns8482e**
PR allows to access named shape in cshtml

```
Model.Content.ThePartName
Model.Content.TheContentType__TheFieldName
```

Fixes #3167

### CODE REVIEW

**Comment by ns8482e** [comment]
Field name requires differentiator like "TheContentType-TheFieldName"
In razor "-" is not valid in property name, hence the field shape is mapped to "TheContentType__TheFieldName"

But liquid template uses differentiator like below
`{{ Model.Content.TheContentType-TheFieldName | shape_render }}`

I feel we should also change the liquid accessor to use shape name using "__ " instead of "-"

**Comment by ns8482e** [comment]
Added Items property to Zones for liquid
Fixes #4553

**Comment by ns8482e** [comment]
Added support for liquid to resolve using __ for field shapes
Now, in liquid
`{{ Model.Content.TheContentType__TheFieldName | shape_render }}`
and 
`{{ Model.Content.TheContentType-TheFieldName | shape_render }}`

both works.

**Comment by sebastienros** [comment]
Shouldn't the change be in the ZoneShape only?

**Comment by ns8482e** [comment]
@sebastienros Yes you are correct, changes for #3167 for cshtml requires only change in Shape.cs
However while working on #4334 for liquid noticed that ShapeAccessor class has following code same as as Named function in Shape.cs - why duplicate it here? So removed following lines and replaced with Named function.

```
                foreach (var item in shape.Items)
                {
                    // Resolve Model.Content.MyNamedPart
                    if (item is IShape itemShape && itemShape.Metadata.Name == n)
                    {
                        return item;
                    }
                }
```

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs [hunk 1]
```diff
@@ -158,14 +158,13 @@ internal class ShapeAccessor : DelegateAccessor
...
-                    // Resolve Model.Content.MyNamedPart
-                    if (item is IShape itemShape && itemShape.Metadata.Name == n)
-                    {
-                        return item;
-                    }
+                    return shape.Items;
                 }
+                // Resolve Model.Content.MyNamedPart
+                // Resolve Model.Content.MyType__MyField OR Resolve Model.Content.MyType-MyField
+                return shape.Named(n.Replace("__", "-"));
```
Comment (ns8482e) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs]: Removed same code as Named function, replaced with Named function

**Pull request merged**